### PR TITLE
Added notes for incompatibilities with certain TensorFlow versions.

### PR DESCRIPTION
**Pull request by Naraenda**
Should be merged/cherry-picked into https://github.com/RadeonOpenCompute/ROCm/pull/2374 and generate a new change log

### CODE REVIEW

File: docs/how_to/tensorflow_install/tensorflow_install.md [hunk 1]
```diff
@@ -9,6 +9,19 @@ training and inference in neural networks. It is one of the most popular and
 in-demand frameworks and is very active in open source contribution and
 development.
 
+:::{warning}
+ROCm 5.6 and 5.7 deviate from the standard practice of supporting the last three
+TensorFlow versions. This is due to incompatibilities between earlier TensorFlow
+versions and changes introduced in the ROCm 5.6 compiler. Please refer to the
```
Comment (Rmalavally) [inline docs/how_to/tensorflow_install/tensorflow_install.md:15]: Please remove the word 'Please' :) Has Aswin's team reviewed this note? Please confirm.

@jithunnair-amd @amathews-amd

File: docs/how_to/tensorflow_install/tensorflow_install.md [hunk 2]
```diff
@@ -9,6 +9,19 @@ training and inference in neural networks. It is one of the most popular and
...
+following version support matrix:
+
+| ROCm  | TensorFlow |
+|:-----:|:----------:|
+| 5.6.x | 2.12       |
+| 5.7.0 | 2.12, 2.13 |
+| Post-5.7.0 | Back to standard support practice. |
```
Comment (Rmalavally) [inline docs/how_to/tensorflow_install/tensorflow_install.md:22]: What is 'standard support practice'? How can users find this information? Can we add a link to best practices?

File: docs/how_to/tensorflow_install/tensorflow_install.md (hunk 1 above)
Comment (amathews-amd) [reply docs/how_to/tensorflow_install/tensorflow_install.md:15]: cc: @jayfurmanek

File: docs/how_to/tensorflow_install/tensorflow_install.md (hunk 2 above)
Comment (Naraenda) [reply docs/how_to/tensorflow_install/tensorflow_install.md]: I have made the description more explicit

File: docs/how_to/tensorflow_install/tensorflow_install.md (hunk 1 above)
Comment (Naraenda) [reply docs/how_to/tensorflow_install/tensorflow_install.md]: Removed :)

**Review by jayfurmanek — APPROVED** [review]

File: docs/how_to/tensorflow_install/tensorflow_install.md (hunk 2 above)
Comment (Rmalavally) [reply docs/how_to/tensorflow_install/tensorflow_install.md]: Thanks so much, Nara.

**Review by saadrahim — APPROVED** [review]

**Pull request merged**